types/Support/Collection.php

```diff
@@ -500,15 +500,15 @@ function ($collection, $count) {
 
 assertType('Illuminate\Support\Collection<string, int>', $collection::make(['string'])->flip());
 
-assertType('Illuminate\Support\Collection<(int|string), Illuminate\Support\Collection<(int|string), User>>', $collection->groupBy('name'));
-assertType('Illuminate\Support\Collection<(int|string), Illuminate\Support\Collection<(int|string), User>>', $collection->groupBy('name', true));
-assertType('Illuminate\Support\Collection<(int|string), Illuminate\Support\Collection<(int|string), User>>', $collection->groupBy(function ($user, $int) {
+assertType('Illuminate\Support\Collection<(int|string), Illuminate\Support\Collection<int, User>>', $collection->groupBy('name'));
```

Comment:
`$collection` `TKey` is explicitly an `int`, see `assertType('Illuminate\Support\Collection<int, User>', $collection);` line 25. So it makes sense that the key type is not broadening when we use groupBy().